lib/jekyll/frontmatter_defaults.rb

```diff
@@ -129,6 +130,14 @@ def path_is_subpath?(path, parent_path)
       false
     end
 
+    def strip_collections_dir(path)
+      collections_dir  = @site.config["collections_dir"]
+      slashed_coll_dir = "#{collections_dir}/"
+      return path if collections_dir.empty?
```

Comment:
I feel like this and the following line could be consolidated into one block to clear up confusion.